src/WorkOS.net/Services/MFA/MFAService.cs

```diff
@@ -78,7 +78,23 @@ public MfaService(WorkOSClient client)
                 Method = HttpMethod.Post,
                 Path = "/auth/factors/verify",
             };
-            return await this.Client.MakeAPIRequest<VerifyFactorResponse>(request, cancellationToken);
+            var response = await this.Client.MakeAPIRequest<VerifyFactorResponse>(request, cancellationToken);
+
+            if (response.Challenge is null && response.IsValid is false && response.ErrorCode.Equals("authentication_challenge_previously_verified"))
```

Comment:
The [`is` operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/is) in C# is used for type testing, so it's not really idiomatic to use it for booleans (even if it does work):

```suggestion
            if (response.Challenge is null && !response.IsValid && response.ErrorCode.Equals("authentication_challenge_previously_verified"))
```